frontend/public/components/utils/file-input.tsx

```diff
@@ -1,50 +1,151 @@
+import { DropTarget, ConnectDropTarget, DropTargetMonitor } from 'react-dnd';
+import withDragDropContext from '../utils/drag-drop-context';
 
 export class FileInput extends React.Component<FileInputProps, FileInputState> {
   constructor(props) {
     super(props);
-    this.state = {
-      inputFileName: '',
-      inputFileData: '',
-    };
-    this.onFileChange = this.onFileChange.bind(this);
+    this.onDataChange = this.onDataChange.bind(this);
+    this.onFileUpload = this.onFileUpload.bind(this);
+  }
+  onDataChange(event) {
+    this.props.onChange(event.target.value);
   }
-  onFileChange(event) {
-    const file = event.target.files[0];
-    this.setState({ inputFileName: file.name });
+  readFile(file) {
     const reader = new FileReader();
     reader.onload = () => {
       const input = reader.result;
-      this.setState({
-        inputFileData: input
-      }, () => this.props.onChange(this.state.inputFileData));
+      this.props.onChange(input, file.name);
     };
     reader.readAsText(file, 'UTF-8');
   }
+  onFileUpload(event) {
+    this.readFile(event.target.files[0]);
+  }
   render() {
-    return <div className="input-group">
-      <input type="text"
-        className="form-control"
-        value={this.state.inputFileName}
-        readOnly
-        disabled />
-      <span className="input-group-btn">
-        <span className="btn btn-default co-btn-file">
-          Browse&hellip;
-          <input type="file" onChange={this.onFileChange} className="form-control"/>
-        </span>
-      </span>
-    </div>;
+    const id = this.props.id;
+    const { connectDropTarget, isOver, canDrop } = this.props;
+    return (
+      connectDropTarget &&
+      connectDropTarget(
+        <div className="co-file-dropzone">
+          { canDrop && <div className={isOver ? 'co-file-dropzone-container co-file-dropzone__drop-over' : 'co-file-dropzone-container'}><p className="co-file-dropzone__drop-text">Drop file here</p></div> }
+          <div className="form-group">
+            <label className="control-label" htmlFor={id}>{this.props.label}</label>
+            <div className="modal-body__field">
+              <div className="input-group">
+                <input type="text"
+                  className="form-control"
+                  value={this.props.inputFileName}
+                  aria-describedby={`${id}-help`}
+                  readOnly
+                  disabled />
+                <span className="input-group-btn">
+                  <span className="btn btn-default co-btn-file">
+                    Browse&hellip;
+                    <input type="file" onChange={this.onFileUpload} className="form-control"/>
+                  </span>
+                </span>
+              </div>
+              <p className="help-block" id={`${id}-help`}>{this.props.inputFieldHelpText}</p>
+              <textarea className="form-control co-create-secret-form__textarea"
+                onChange={this.onDataChange}
```

Comment:
Yeah was thinking about this, but figured out that it could be handy to see the which file was uploaded, and even if user decides to change some parts of the uploaded file, he will still see which file he is based from